### PR TITLE
fix: 🐛 type error in icestark

### PR DESCRIPTION
**Pull request by maoxiaoke**
✅ Closes: #4800, #4833

### CODE REVIEW

File: packages/plugin-icestark/package.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
   },
   "devDependencies": {
     "@types/glob": "^7.1.1",
+    "@types/react-dom": "16",
```
Comment (ClarkXia) [inline packages/plugin-icestark/package.json:29]: 考虑直接 17 吧，对外默认的模版都升级到 17 了

Comment (maoxiaoke) [reply packages/plugin-icestark/package.json]: done

File: packages/plugin-icestark/src/runtime.tsx [hunk 2]
```diff
@@ -59,7 +61,7 @@ const module = ({
               if (enterRegistration) {
                 enterRegistration(mountNode, App, resolve);
               } else {
-                ReactDOM.render(<App />, mountNode, resolve);
+                ReactDOM.render(<App />, mountNode, resolve as Noop);
```
Comment (ClarkXia) [inline packages/plugin-icestark/src/runtime.tsx:64]: () => {
  resolve();
}

Comment (maoxiaoke) [reply packages/plugin-icestark/src/runtime.tsx]: done

**Review by ClarkXia — APPROVED** [review]

**Pull request merged**